mlrun/kfpops.py

```diff
@@ -821,3 +825,38 @@ def add_function_node_selection_attributes(
             container_op.affinity = function.spec.affinity
 
     return container_op
+
+
+def add_function_security_context(
+    function, container_op: dsl.ContainerOp
+) -> dsl.ContainerOp:
+
+    if (
+        mlrun.runtimes.RuntimeKinds.is_local_runtime(function.kind)
+        or mlrun.mlconf.function.spec.security_context.enrichment_mode
+        == SecurityContextEnrichmentModes.disabled.value
+    ):
+        return container_op
+
+    # ensure kfp pod user id is not None or 0 (root)
+    if not mlrun.mlconf.function.spec.security_context.pipelines.kfp_pod_user_unix_id:
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            f"Kubeflow pipeline pod user id is invalid: "
+            f"{mlrun.mlconf.function.spec.security_context.pipelines.kfp_pod_user_unix_id}, "
+            f"it must be an integer greater than 0. "
+            f"See config.function.spec.security_context.pipelines.kfp_pod_user_unix_id for more details."
+        )
+
+    kfp_pod_user_unix_id = int(
+        mlrun.mlconf.function.spec.security_context.pipelines.kfp_pod_user_unix_id
+    )
+    container_op.container.set_security_context(
+        k8s_client.V1SecurityContext(
+            run_as_user=kfp_pod_user_unix_id,
+            run_as_group=mlrun.mlconf.get_security_context_enrichment_group_id(
+                kfp_pod_user_unix_id
+            ),
+        )
+    )
```

Comment:
is this required if you are using `_create_enriched_mlrun_workflow` to enrich the kfp pods?

Reply:
moved this logic to where we enrich the pod 👍